zellij-server/src/panes/tiled_panes/mod.rs

```diff
@@ -1020,7 +1036,12 @@ impl TiledPanes {
                     *self.display_area.borrow(),
                     *self.viewport.borrow(),
                 );
-                let next_index = pane_grid.next_selectable_pane_id_to_the_right(&active_pane_id);
+                let mut next_index =
+                    pane_grid.next_selectable_pane_id_to_the_right(&active_pane_id);
+
+                if next_index.is_none() && single_tab_wraparoud {
+                    next_index = Some(PaneId::Terminal(0));
```

Comment:
Same here as above - I don't think this will always work. IMO it would be best to explicitly search for the leftmost/rightmost pane with `TiledPaneGrid` and `FloatingPaneGrid`

Reply:
Repeat `next_selectable_pane_id_to_the_right` to update `Id` until Option -> None? Does this sound right ?